zkevm-circuits/src/evm_circuit/witness.rs

```diff
@@ -0,0 +1,520 @@
+            push_data_left: 0,
+            hash: RandomLinearCombination::random_linear_combine(
+                self.hash.to_le_bytes(),
+                randomness,
+            ),
+            bytes: &self.bytes,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum Rw {
+    TxAccessListAccount {
+        rw_counter: usize,
+        is_write: bool,
+    },
+    TxAccessListStorageSlot {
+        rw_counter: usize,
+        is_write: bool,
+    },
+    TxRefund {
+        rw_counter: usize,
+        is_write: bool,
+    },
+    Account {
+        rw_counter: usize,
+        is_write: bool,
+    },
+    AccountStorage {
+        rw_counter: usize,
+        is_write: bool,
+    },
+    AccountDestructed {
+        rw_counter: usize,
+        is_write: bool,
+    },
+    CallContext {
+        rw_counter: usize,
+        is_write: bool,
+    },
+    Stack {
+        rw_counter: usize,
+        is_write: bool,
+        call_id: usize,
+        stack_pointer: usize,
+        value: Word,
+    },
+    Memory {
+        rw_counter: usize,
+        is_write: bool,
+        call_id: usize,
+        memory_address: u64,
+        byte: u8,
+    },
+}
+
+impl Rw {
+    pub fn stack_value(&self) -> Word {
+        match self {
+            Self::Stack { value, .. } => *value,
+            _ => unreachable!(),
+        }
+    }
+
+    pub fn table_assignment<F: FieldExt>(&self, randomness: F) -> [F; 8] {
+        match self {
+            Self::Stack {
+                rw_counter,
+                is_write,
+                call_id,
+                stack_pointer,
+                value,
+            } => [
+                F::from(*rw_counter as u64),
+                F::from(*is_write as u64),
+                F::from(RwTableTag::Stack as u64),
+                F::from(*call_id as u64),
+                F::from(*stack_pointer as u64),
+                RandomLinearCombination::random_linear_combine(
+                    value.to_le_bytes(),
+                    randomness,
+                ),
+                F::zero(),
+                F::zero(),
+            ],
+            Self::Memory {
+                rw_counter,
+                is_write,
+                call_id,
+                memory_address,
+                byte,
+            } => [
+                F::from(*rw_counter as u64),
+                F::from(*is_write as u64),
+                F::from(RwTableTag::Memory as u64),
+                F::from(*call_id as u64),
+                F::from(*memory_address),
+                F::from(*byte as u64),
+                F::zero(),
+                F::zero(),
+            ],
+            _ => unimplemented!(),
+        }
+    }
+}
+
+use halo2::arithmetic::BaseExt;
+use pairing::bn256::Fr as Fp;
+use std::convert::TryInto;
+
+impl From<&bus_mapping::circuit_input_builder::ExecStep> for ExecutionState {
+    fn from(step: &bus_mapping::circuit_input_builder::ExecStep) -> Self {
+        // TODO: convert error (circuit_input_builder.rs)
```

Comment:
rephrase

Reply:
addressed in https://github.com/scroll-tech/zkevm-circuits/pull/30/commits/3184aec2ffa9395a6b2af729c76249dd65d0a899